deduplication-document-spark/deduplication-document-spark-impl/src/main/scala/pl/edu/icm/coansys/document/deduplication/MultiLengthTitleKeyGenerator.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of CoAnSys project.
+ * Copyright (c) 2012-2017 ICM-UW
+ * 
+ * CoAnSys is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * CoAnSys is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Affero General Public License
+ * along with CoAnSys. If not, see <http://www.gnu.org/licenses/>.
+ */
+package pl.edu.icm.coansys.document.deduplication
+import pl.edu.icm.coansys.commons.java.DocumentWrapperUtils
+import pl.edu.icm.coansys.commons.java.StringTools
+import pl.edu.icm.coansys.models.DocumentProtos.DocumentMetadata
+
+/**
+ * Generator for the keys used in early stage of the document deduplication.
+ */
+class MultiLengthTitleKeyGenerator(val keySizes: Seq[Int]) {
+  def cleanUpString(title: String): String = {
+    val normalized = StringTools.normalize(title);
+    //seems that normalize removes stopwords, which is wrong, and quite expensive
+    //val normalized = StringTools.removeStopWords(StringTools.normalize(title));
+    val res = normalized.replaceAll("\\s+", "")
```

Comment:
There is no need to define `val res` here, the expression assigned to it could just be simply the last expression in this function.